File: frontend/src/data/cities.ts
export interface City {
  id: string;
  name: string;
  quartiers?: string[];
}

export const cities: City[] = [
  {
    id: "ndjamena",
    name: "N'Djaména",
    quartiers: [
      "Moursal", "Chagoua", "Dembé", "Farcha", "Diguel",
      "Amriguébé", "Ardep Djoumal", "Bololo", "Djambal Bahr",
      "Gardolé", "Habena", "Klémat", "Mardjandafack",
      "Ndjari", "Paris-Congo", "Sabangali", "Ridina",
      "Amtoukoui", "Gassi", "Nguéli"
    ],
  },
  { id: "moundou", name: "Moundou" },
  { id: "sarh", name: "Sarh" },
  { id: "abeche", name: "Abéché" },
  { id: "kelo", name: "Kélo" },
  { id: "koumra", name: "Koumra" },
  { id: "pala", name: "Pala" },
  { id: "amtiman", name: "Am-Timan" },
  { id: "bongor", name: "Bongor" },
  { id: "mongo", name: "Mongo" },
  { id: "doba", name: "Doba" },
  { id: "ati", name: "Ati" },
  { id: "mao", name: "Mao" },
];

export function getCityById(id: string): City | undefined {
  return cities.find((c) => c.id === id);
}

File: mobile/lib/data/cities.dart
class CityData {
  const CityData({required this.id, required this.name, this.quartiers = const []});
  final String id;
  final String name;
  final List<String> quartiers;
}

const cities = <CityData>[
  CityData(
    id: 'ndjamena',
    name: "N'Djaména",
    quartiers: [
      'Moursal', 'Chagoua', 'Dembé', 'Farcha', 'Diguel',
      'Amriguébé', 'Ardep Djoumal', 'Bololo', 'Djambal Bahr',
      'Gardolé', 'Habena', 'Klémat', 'Mardjandafack',
      'Ndjari', 'Paris-Congo', 'Sabangali', 'Ridina',
      'Amtoukoui', 'Gassi', 'Nguéli',
    ],
  ),
  CityData(id: 'moundou', name: 'Moundou'),
  CityData(id: 'sarh', name: 'Sarh'),
  CityData(id: 'abeche', name: 'Abéché'),
  CityData(id: 'kelo', name: 'Kélo'),
  CityData(id: 'koumra', name: 'Koumra'),
  CityData(id: 'pala', name: 'Pala'),
  CityData(id: 'amtiman', name: 'Am-Timan'),
  CityData(id: 'bongor', name: 'Bongor'),
  CityData(id: 'mongo', name: 'Mongo'),
  CityData(id: 'doba', name: 'Doba'),
  CityData(id: 'ati', name: 'Ati'),
  CityData(id: 'mao', name: 'Mao'),
];

CityData? getCityById(String id) {
  for (final c in cities) {
    if (c.id == id) return c;
  }
  return null;
}
